codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/knowledge/GoPointableIndex.java

```diff
@@ -15,263 +15,270 @@
-    }
-
-    private boolean isShapePointable(Shape shape) {
-        // All operation input and output shapes are pointable.
-        if (isOperationStruct(shape)) {
-            return true;
-        }
-
-        // Streamed blob shapes are never pointers because they are interfaces
-        if (isBlobStream(shape)) {
-            return false;
-        }
-
-        if (shape.isServiceShape()) {
-            return true;
-        }
-
-        // This is odd because its not a go type but a function with receiver
-        if (shape.isOperationShape()) {
-            return false;
-        }
-
-        if (INHERENTLY_POINTABLE.contains(shape.getType())) {
-            return true;
-        }
-
-        if (INHERENTLY_VALUE.contains(shape.getType())) {
-            return false;
-        }
-
-        if (shape.isIntegerShape() && !shape.hasTrait(RequiredTrait.class)) {
-            return true;
-        }
-
-        return nullableIndex.isNullable(shape);
-    }
-
-    private boolean isShapeEnum(Shape shape) {
-        return shape.getType() == ShapeType.STRING && shape.hasTrait(EnumTrait.class)
-                || shape.getType() == ShapeType.ENUM
-                || shape.getType() == ShapeType.INT_ENUM;
-    }
-
-    private boolean isBlobStream(Shape shape) {
-        return shape.getType() == ShapeType.BLOB && shape.hasTrait(StreamingTrait.ID);
-    }
-
-    private boolean isOperationStruct(Shape shape) {
-        NeighborProvider provider = NeighborProviderIndex.of(model).getReverseProvider();
-        for (Relationship relationship : provider.getNeighbors(shape)) {
-            RelationshipType relationshipType = relationship.getRelationshipType();
-            if (relationshipType == RelationshipType.INPUT || relationshipType == RelationshipType.OUTPUT) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    /**
-     * Returns if the shape should be generated as a Go pointer type or not.
-     *
-     * @param shape the shape to check if should be pointable type.
-     * @return if the shape is should be a Go pointer type.
-     */
-    public final boolean isPointable(ToShapeId shape) {
-        return pointableShapes.contains(shape.toShapeId());
-    }
-
-    /**
-     * Returns if the Go type generated for the shape is comparable to nil.
-     *
-     * @param shape the shape to check
-     * @return if the shape's go type is comparable to nil
-     */
-    public final boolean isNillable(ToShapeId shape) {
-        return nillableShapes.contains(shape.toShapeId());
-    }
-
-    /**
-     * Returns if the Go type generated for the shape can be dereferenced.
-     *
-     * @param shape the shape to check
-     * @return if the shape's go type is dereferencable
-     */
-    public final boolean isDereferencable(ToShapeId shape) {
-        return dereferencableShapes.contains(shape.toShapeId());
-    }
-}
+ package software.amazon.polymorph.smithygo.codegen.knowledge;
```

Comment:
What happened here? Why the whole file is in the diff??

Reply:
This diff will go away once I merge new changes to this PR.